bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/Configuration.java

```diff
@@ -119,6 +127,38 @@ public static String getGdbServerCommandLine(ILaunchConfiguration configuration)
 		return lst.toArray(new String[0]);
 	}
 
+	private static boolean checkIfJtagIsAvailable()
+	{
+		EspConfigParser parser = new EspConfigParser();
+		String openOCDPath = new IDFEnvironmentVariables().getEnvValue(IDFEnvironmentVariables.OPENOCD_SCRIPTS);
+		if (!openOCDPath.isEmpty() && parser.hasBoardConfigJson())
+		{
+			return true;
+		}
+		return false;
+	}
+
+	private static String getEspFlashCommand(ILaunchConfiguration configuration)
+	{
+		String espFlashCommand = "program_esp_bins <path-to-build-dir> flasher_args.json verify reset"; //$NON-NLS-1$
+		try
+		{
+			String buildPath = configuration.getMappedResources()[0].getProject().getFolder("build").getLocationURI() // $NON-NLS-1$
+					.getPath();
+			char a = buildPath.charAt(2);
+			if (a == ':')
+			{
+				buildPath = buildPath.substring(1);
+			}
+			espFlashCommand = espFlashCommand.replace("<path-to-build-dir>", buildPath); //$NON-NLS-1$
+		}
+		catch (CoreException e)
+		{
+			Logger.log(e);
+		}
+		return espFlashCommand;
+	}
+
```

Comment:
I think this was also used during JTAG flashing implementation, can we move this to a common place?

Reply:
@kolipakakondal moved to ESPFlashUtil